### PR TITLE
Ruleset

### LINKED ISSUE #67: How do we name the platform configuration?
https://strawpoll.com/k5pr2pyz

### PR DESCRIPTION
**Pull request by lachmanfrantisek**
- Renamed config to ruleset
- Allow set ruleset from `COLIN_RULESET` environment variable.
- Corrected README.
- Implement `list-rulesets`

Resolves #67

### CODE REVIEW

File: colin/core/ruleset/ruleset.py [hunk 1]
```diff
@@ -162,47 +162,47 @@ def get_checks_path():
...
 
-    config_directory = get_config_directory()
-    config_file = os.path.join(config_directory, config + JSON)
+    ruleset_directory = get_ruleset_directory()
+    ruleset_file = os.path.join(ruleset_directory, ruleset + JSON)
 
-    if os.path.exists(config_file) and os.path.isfile(config_file):
-        logger.debug("Configuration file '{}' found.".format(config_file))
-        return config_file
+    if os.path.exists(ruleset_file) and os.path.isfile(ruleset_file):
```
Comment (jpopelka) [inline colin/core/ruleset/ruleset.py]: Isn't `os.path.exists(ruleset_file)` superset of `os.path.isfile(ruleset_file)` ?
I mean, I think just
```python
if os.path.isfile(ruleset_file):
```
should do the same.

File: colin/core/ruleset/ruleset.py [hunk 2]
```diff
@@ -162,47 +162,47 @@ def get_checks_path():
...
+    First directory to check:  $HOME/.local/share/colin/ruleset
+    Second directory to check: /usr/local/share/colin/ruleset
     :return: str
     """
 
     local_share = os.path.join(os.path.expanduser("~"),
                                ".local",
-                               CONFIG_DIRECTORY)
+                               RULESET_DIRECTORY)
     if os.path.isdir(local_share) and os.path.exists(local_share):
```
Comment (jpopelka) [inline colin/core/ruleset/ruleset.py]: I think just `if os.path.isdir(local_share)` should be the same.

File: colin/core/ruleset/ruleset.py [hunk 3]
```diff
@@ -162,47 +162,47 @@ def get_checks_path():
...
-        logger.debug("Local configuration directory found ('{}').".format(local_share))
+        logger.debug("Local ruleset directory found ('{}').".format(local_share))
         return local_share
 
-    usr_local_share = os.path.join("/usr/local", CONFIG_DIRECTORY)
+    usr_local_share = os.path.join("/usr/local", RULESET_DIRECTORY)
     if os.path.isdir(usr_local_share) and os.path.exists(usr_local_share):
```
Comment (jpopelka) [inline colin/core/ruleset/ruleset.py]: Or just `if os.path.isdir(usr_local_share):` ?

File: README.md [hunk 4]
```diff
@@ -34,22 +34,23 @@ This is how you can use colin afterwards:
...
+Usage: colin [OPTIONS] COMMAND [ARGS]...
+
+  COLIN -- Container Linter
 
 Options:
-  -c, --config [redhat|fedora]  Select a predefined configuration.
-  --debug                       Enable debugging mode (debugging logs, full tracebacks).
-  -f, --config-file FILENAME    Path to a file to use for validation (by
-                                default they are placed in /usr/share/colin).
-  --json FILENAME               File to save the output as json to.
```
Comment (phracek) [inline README.md]: Why did you remove --json option?

File: README.md [hunk 5]
```diff
@@ -34,22 +34,23 @@ This is how you can use colin afterwards:
...
-  -s, --stat                    Print statistics instead of full results.
```
Comment (phracek) [inline README.md]: The same as above.

**Comment by phracek** [comment]
Otherwise. LGTM.

File: README.md (hunk 4 above)
Comment (lachmanfrantisek) [reply README.md]: The `--json` is only valid for subcommands. (`check`/`list-checks`)

**Comment by lachmanfrantisek** [comment]
@phracek  I've added the help output for the `colin check`.

(The `check` is a default subcommand, so you can use `colin check my-fedora` or `colin my-fedora`, but the help messages are different.)

File: colin/core/ruleset/ruleset.py [hunk 6]
```diff
@@ -20,45 +20,45 @@
...
+                raise ColinRulesetException(msg)
         else:
             try:
-                logger.debug("Loading configuration with the name '{}'.".format(config_name))
-                config_path = config_file or get_config_file(config=config_name)
-                with open(config_path, mode='r') as config_file_obj:
-                    self.config_dict = json.load(config_file_obj)
-            except ColinConfigException as ex:
+                logger.debug("Loading ruleset with the name '{}'.".format(ruleset_name))
+                ruleset_path = ruleset_file or get_ruleset_file(ruleset=ruleset_name)
```
Comment (jpopelka) [inline colin/core/ruleset/ruleset.py]: I think `ruleset_file` is always `None` at this point, because if it was something else we'd not end up in this `else:` branch, but in the above `if ruleset_file:`. What I'm suggesting is
```python
ruleset_path = get_ruleset_file(ruleset=ruleset_name)
```

File: colin/core/ruleset/ruleset.py [hunk 7]
```diff
@@ -20,45 +20,45 @@
...
-    def __init__(self, config_name=None, config_file=None):
+    def __init__(self, ruleset_name=None, ruleset_file=None):
         """
-        Load config for colin.
+        Load ruleset for colin.
 
-        :param config_name: str (name of the config file (without .json), default is "default"
-        :param config_file: fileobj
+        :param ruleset_name: str (name of the ruleset file (without .json), default is "default"
+        :param ruleset_file: fileobj
```
Comment (jpopelka) [inline colin/core/ruleset/ruleset.py]: `ruleset_file` looks more like a str with a path to me. Am I missing something ?

File: colin/core/ruleset/ruleset.py (hunk 1 above)
Comment (lachmanfrantisek) [reply colin/core/ruleset/ruleset.py]: It is not something related to the pull request, but yes, you're right.

File: colin/core/ruleset/ruleset.py (hunk 7 above)
Comment (lachmanfrantisek) [reply colin/core/ruleset/ruleset.py]: No, it's really the file object. (click checks the permission and gives us the file object)

Comment (lachmanfrantisek) [reply colin/core/ruleset/ruleset.py]: Good point. Corrected.

File: setup.py [hunk 8]
```diff
@@ -27,9 +27,9 @@
         [console_scripts]
         colin=colin.cli.colin:cli
     ''',
-    data_files=[("share/colin/config/", ["config/default.json",
-                                         "config/fedora.json",
-                                         "config/redhat.json"]),
+    data_files=[("share/colin/rulesets/", ["ruleset/default.json",
+                                           "ruleset/fedora.json",
+                                           "ruleset/redhat.json"]),
```
Comment (jpopelka) [inline setup.py]: Actually, wouldn't it be better if the directory in the repo is also called `rulesets` (not `ruleset`) ?

**Comment by lachmanfrantisek** [comment]
Sorry to change the code in the progress, but I've added the implementation of the `list-resultsets` as well.

File: setup.py (hunk 8 above)
Comment (lachmanfrantisek) [reply setup.py]: Yes, I've changed that as well.

File: colin/core/ruleset/ruleset.py [hunk 9]
```diff
@@ -206,3 +206,13 @@ def get_ruleset_directory():
...
     raise ColinRulesetException(msg)
+
+
+def get_rulesets():
+    """"
+    Get available rulesets.
+    """
+    rulesets_dir = get_ruleset_directory()
+    ruleset_files = [os.path.basename(f)[:-5] for f in os.listdir(rulesets_dir) if
+                     os.path.isfile(os.path.join(rulesets_dir, f)) and f.lower().endswith(JSON)]
```
Comment (jpopelka) [inline colin/core/ruleset/ruleset.py]: Hehe, I've just implemented it myself, but you've been quicker.
Anyway I'd prefer `len(JSON)` if that's what the `5` means.
Also the `basename()` is I think not necessary, but does no harm, so OK to me.

Comment (lachmanfrantisek) [reply colin/core/ruleset/ruleset.py]: Sorry..;-) (I will use that, so it will be more readable.)

Comment (jpopelka) [reply colin/core/ruleset/ruleset.py]: LGTM.
BTW, do you know [glob](https://docs.python.org/3/library/glob.html#glob.glob) ?
```
ruleset_files = [os.path.basename(f)[:-len(JSON)].lower() for f in glob.glob(rulesets_dir + '/*' + JSON)]
```

**Comment by lachmanfrantisek** [comment]
Thank both of you for review(s).
